lib/rubocop/cop/factory_bot/id_sequence.rb

```diff
@@ -19,12 +19,15 @@ module FactoryBot
       class IdSequence < ::RuboCop::Cop::Base
         extend AutoCorrector
         include RangeHelp
+        include RuboCop::FactoryBot::Language
 
         MSG = 'Do not create a sequence for an id attribute'
         RESTRICT_ON_SEND = %i[sequence].freeze
 
         def on_send(node)
-          return unless node.first_argument.value == :id
+          return unless node.receiver.nil? || factory_bot?(node.receiver)
```

Comment:
doesn't factory_bot?(node.receiver) handle the case when receiver is nil?

Reply:
It appears so.
```
RuboCop::Cop::FactoryBot::IdSequence
  does not register an offense for a `sequence` with receiver
  registers an offense with a Enumerable of values (FAILED - 1)
  does not register an offense for a `sequence` without argument
  registers an offense with no block (FAILED - 2)
  does not register an offense for a non-id sequence
  registers an offense with a default value (FAILED - 3)
  does not register an offense for a `sequence` with non-symbol argment
  registers an offense across multiple lines (FAILED - 4)

Failures:

  1) RuboCop::Cop::FactoryBot::IdSequence registers an offense with a Enumerable of values
     Failure/Error:
       expect_offense(<<~RUBY)
         FactoryBot.define do
           factory :post do
             summary { "A summary" }
             sequence(:id, (1..10).cycle)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not create a sequence for an id attribute
             title { "A title" }
           end
         end
       RUBY
     
       Diff:
       @@ -2,7 +2,6 @@
          factory :post do
            summary { "A summary" }
            sequence(:id, (1..10).cycle)
       -    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not create a sequence for an id attribute
            title { "A title" }
          end
        end
       
     # ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:74:in `block (2 levels) in <top (required)>'

  2) RuboCop::Cop::FactoryBot::IdSequence registers an offense with no block
     Failure/Error:
       expect_offense(<<~RUBY)
         FactoryBot.define do
           factory :post do
             summary { "A summary" }
             sequence :id
             ^^^^^^^^^^^^ Do not create a sequence for an id attribute
             title { "A title" }
           end
         end
       RUBY
     
       Diff:
       @@ -2,7 +2,6 @@
          factory :post do
            summary { "A summary" }
            sequence :id
       -    ^^^^^^^^^^^^ Do not create a sequence for an id attribute
            title { "A title" }
          end
        end
       
     # ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:5:in `block (2 levels) in <top (required)>'

  3) RuboCop::Cop::FactoryBot::IdSequence registers an offense with a default value
     Failure/Error:
       expect_offense(<<~RUBY)
         FactoryBot.define do
           factory :post do
             summary { "A summary" }
             sequence(:id, 1000)
             ^^^^^^^^^^^^^^^^^^^ Do not create a sequence for an id attribute
             title { "A title" }
           end
         end
       RUBY
     
       Diff:
       @@ -2,7 +2,6 @@
          factory :post do
            summary { "A summary" }
            sequence(:id, 1000)
       -    ^^^^^^^^^^^^^^^^^^^ Do not create a sequence for an id attribute
            title { "A title" }
          end
        end
       
     # ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:27:in `block (2 levels) in <top (required)>'

  4) RuboCop::Cop::FactoryBot::IdSequence registers an offense across multiple lines
     Failure/Error:
       expect_offense(<<~RUBY)
         FactoryBot.define do
           factory :post do
             summary { "A summary" }
             sequence(
             ^^^^^^^^^ Do not create a sequence for an id attribute
               :id,
               1000
             )
             title { "A title" }
     
       Diff:
       @@ -2,7 +2,6 @@
          factory :post do
            summary { "A summary" }
            sequence(
       -    ^^^^^^^^^ Do not create a sequence for an id attribute
              :id,
              1000
            )
       
     # ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:49:in `block (2 levels) in <top (required)>'

Finished in 0.77296 seconds (files took 1.56 seconds to load)
289 examples, 4 failures

Failed examples:

rspec ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:73 # RuboCop::Cop::FactoryBot::IdSequence registers an offense with a Enumerable of values
rspec ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:4 # RuboCop::Cop::FactoryBot::IdSequence registers an offense with no block
rspec ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:26 # RuboCop::Cop::FactoryBot::IdSequence registers an offense with a default value
rspec ./spec/rubocop/cop/factory_bot/id_sequence_spec.rb:48 # RuboCop::Cop::FactoryBot::IdSequence registers an offense across multiple lines
```

Reply:
Ah, I see
The ConfigurableExplicitOnly handles it as
```
def factory_call?(node)
          return factory_bot?(node) if explicit_only?

          factory_bot?(node) || node.nil? # NOTE the nil check
        end
```

Reply:
ah I expected the definition of factory_bot? to be
```
def_node_matcher :factory_bot?, <<~PATTERN
   {nil? (const {nil? cbase} {:FactoryGirl :FactoryBot})}
 PATTERN
 ```
much like it is in rubocop-rspec
```
def_node_matcher :rspec?, '{#explicit_rspec? nil?}'

def_node_matcher :explicit_rspec?, '(const {nil? cbase} :RSpec)'
```